cmd/singularity/build_test.go

```diff
@@ -110,14 +114,22 @@ func TestBuild(t *testing.T) {
 				}
 			}
 
+			// Set a clean image cache for every test
+			imgCacheDir := test.SetCacheDir(t, "")
```

Comment:
since this bit of code is repeated several times, maybe something like this helps:

```
func setupCache(t *testing.T) (*cache.Handle, func()) {
    dir := test.SetCacheDir(t, "")
    h, err := cache.HdlInit(dir)
    if err != nil {
        test.CleanCacheDir(t, dir)
        t.Fatal("...")
    }
    return h, func() { test.CleanCacheDir(t, dir) }
}
```

and then:

```
    h, cleanup := setupCache(t)
    defer cleanup()
    // use h
```